src/app/components/dynamic-list/DynamicList.jsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import DynamicListItem from "./DynamicListItem";
+import PropTypes from "prop-types";
+import Input from "../Input";
+
+const propTypes = {
+    title: PropTypes.string.isRequired,
+    listHeightClass: PropTypes.string,
+    filterPlaceholder: PropTypes.string,
+    noResultsText: PropTypes.string,
+    handleSearchInput: PropTypes.func,
+    listItems: PropTypes.arrayOf(PropTypes.shape(DynamicListItem.propTypes)).isRequired,
+};
+
+const DynamicList = props => {
+    const searchID = `${props.title.replace(/\s+/g, "-").toLowerCase()}-search-content-types`;
+    let list;
+    if (props.listItems.length > 0) {
+        list = (
+            <ul className={`dynamic-list__container ${props.listHeightClass}`}>
+                {props.listItems.map((row, index) => {
+                    return <DynamicListItem key={`dynamic-list-row-${index}`} {...row} />;
+                })}
+            </ul>
+        );
+    } else {
+        list = <div role="alert">{props.noResultsText}</div>;
```

Comment:
It'll fundamentally depend on how urgent we think it is for the user to be notified that no results have appeared, but my initial feeling is that it might be preferable to use `aria-live="polite"` instead, as the I get the impression this wouldn't be a time-critical notification for the user to hear

Reply:
So good spot around the `aria-live` should be using that here rather than role as this is more of a live element. However, would argue this should be assertive and not polite.

The visual user journey is they use the 'search' bar to filter results. As they type they get instant feedback on results. If there are no results mid typing they know to stop typing as typing more characters won't yield a better result. Likewise they get visual feedback as to what character was the last one they typed that meant their were no results which can help indicate a mispelling, and they will know where that mispelling took place (last character they typed).

Now taking a look at the visually impaired user that relies on audio. If see to polite it will wait for the user to go idle before giving feedback. Meaning they will have to type very slowly one character at a time to get any feedback. A user who opts to use the filter will be using it with an idea of who they are looking for so are likely to type fast. e.g. the word 'Jane'. If we had no Janes but a Jared when they type 'n' there would be no results but they would have typed an extra 'e' meaning polite would have been triggered but never gone idle as the user types an 'e' next so they never hear it (with VoiceOver at least this means nothing is announced when on polite, unless typing very very slowly). With assertive as soon as there are no results they will get this instant feedback just like a visual user.

Happy to have further feedback on this as not done an accessible instant results field before. Another option is to delay the filter - but feel that would be a worse user journey for every user type.

Reply:
I think that's fair enough - thanks for the reasoning